esp/src/src-react/hooks/platform.ts

```diff
@@ -66,3 +67,28 @@ export function useLogicalClusters(): [TpLogicalClusterQuery.TpLogicalCluster[]
 
     return [targetClusters, defaultCluster];
 }
+
+export function useModernMode(): {
```

Comment:
Not sure a hook is good for this (given its needed in Dojo and React, maybe just two funcs:
`getModernMode` and `setModernMode`?